packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx

```diff
@@ -132,6 +132,14 @@ export const DataLinkInput = memo(
 
       switch (event.key) {
         case 'Backspace':
+          const precedingChar: string = getCharactersAroundCaret();
+
+          if (precedingChar === '=' || precedingChar === '$') {
```

Comment:
This is only occurring when the first character ($ or =) is deleted, right? Perhaps it would be easier to just check for string length?

Reply:
Thank you for the suggestion. I agree that checking the string length would simplify the logic. I will make the necessary changes to implement this approach.

Reply:
@drew08t how about using a Boolean variable to solve this easily instead of using string length?